pydantic/_internal/_generate_schema.py

```diff
@@ -346,10 +349,12 @@ class GenerateSchema:
 
     def __init__(
         self,
+        type_registry: TypeRegistry,
```

Comment:
What's the advantage of having this as a field on the instance rather than a module-level import? Do users likely want to use a custom type registry, or is it for us for tests?